proposals/lip-0051.md

```diff
@@ -2635,6 +2635,35 @@ This section specifies the non-trivial or recommended endpoints of the Token mod
 
 This function returns token IDs for both native and non-native tokens supported by the chain.
 
+```python
+def getSupportedTokens() -> dict[list]:
+    # Native tokens
+    nativeTokenIDs = [tokenID for tokenID in supplyStore.keys()]
+
+    # Non-native tokens	
+    if supportedTokens(ALL_SUPPORTED_TOKENS_KEY) exists:
+        return {
+            "nativeTokenIDs": nativeTokenIDs,
+            "supportedChainIDs": "All",
+            "supportedTokenIDs": "All"
+        }
+
+    supportedChainIDs = []
+    supportedTokenIDs = [getMainchainID()]
```

Comment:
In order to be more aligned with the SDK I would write `supportedTokenIDs = [getMainchainID() + b'\x00'*4]` here. The implementation pads the mainchain ID with zeros to the length of a token ID to get the token ID of LSK in the current network.